### PR TITLE
Key manager tests

### PR DESCRIPTION
**Pull request by CJentzsch**
to be continued...

### CODE REVIEW

File: ethkey/KeyAux.h [hunk 1]
```diff
@@ -198,7 +198,16 @@ class KeyCLI
 			if (m_masterPassword.empty())
 				cerr << "Aborted (empty password not allowed)." << endl;
 			else
-				wallet.create(m_masterPassword);
+			{
+				try
+				{
+					wallet.create(m_masterPassword);
+				}
+				catch(Exception const& _e)
```
Comment (chriseth) [inline ethkey/KeyAux.h]: space

File: test/libethcore/keymanager.cpp [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	km.setKeysFile(tmpDir.path() + "keysFile.json");
+	km.create(password);
+	km.save(password);
+
+	BOOST_CHECK(!km.haveHint(password + "2"));
+	km.notePassword(password);
+	BOOST_CHECK(km.haveHint(password));
+}
+
+
```
Comment (winsvega) [inline test/libethcore/keymanager.cpp]: empty lines

File: test/libethcore/keymanager.cpp [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+using namespace std;
+using namespace dev;
+using namespace dev::eth;
+
+BOOST_AUTO_TEST_SUITE(KeyManagerTests)
+
+BOOST_AUTO_TEST_CASE(KeyInfoDefaultConstructor)
+{
+	cnote << "KeyInfoDefaultConstructor";
```
Comment (chriseth) [inline test/libethcore/keymanager.cpp]: Shouldn't that be provided by the test framework?

Comment (CJentzsch) [inline test/libethcore/keymanager.cpp]: When executing the tests without it, it does not show the name.

Comment (chriseth) [inline test/libethcore/keymanager.cpp]: Yes and that is a good thing.

**Comment by gavofyork** [comment]
windows build failed

**Comment by CJentzsch** [comment]
the windos build fail was caused by another PR, here is the fix; https://github.com/ethereum/cpp-ethereum/pull/2272

**Pull request merged**